src/main/java/it/smartcommunitylab/aac/webauthn/service/WebAuthnUserAccountService.java

```diff
@@ -0,0 +1,167 @@
+package it.smartcommunitylab.aac.webauthn.service;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import com.yubico.webauthn.data.ByteArray;
+
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+
+import it.smartcommunitylab.aac.common.NoSuchUserException;
+import it.smartcommunitylab.aac.common.RegistrationException;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredential;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnCredentialsRepository;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccount;
+import it.smartcommunitylab.aac.webauthn.persistence.WebAuthnUserAccountRepository;
+
+@Service
+@Transactional
+public class WebAuthnUserAccountService {
+
+    private final WebAuthnUserAccountRepository accountRepository;
+
+    private final WebAuthnCredentialsRepository credentialRepository;
+
+    public WebAuthnUserAccountService(WebAuthnUserAccountRepository accountRepository,
+            WebAuthnCredentialsRepository credentialRepository) {
+        Assert.notNull(accountRepository, "accountRepository is mandatory");
+        Assert.notNull(credentialRepository, "credentialRepository is mandatory");
+        this.accountRepository = accountRepository;
+        this.credentialRepository = credentialRepository;
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount getAccount(long id) throws NoSuchUserException {
+        WebAuthnUserAccount account = accountRepository.findOne(id);
+        if (account == null) {
+            throw new NoSuchUserException();
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public List<WebAuthnUserAccount> getAccounts(String userId) throws NoSuchUserException {
+        List<WebAuthnUserAccount> accounts = accountRepository.findBySubject(userId);
+        List<WebAuthnUserAccount> result = new LinkedList<>();
+        for (WebAuthnUserAccount account : accounts) {
+            result.add(accountRepository.detach(account));
+        }
+        if (result.isEmpty()) {
+            throw new NoSuchUserException();
+        }
+        return result;
+    }
+
+    @Transactional(readOnly = true)
+    public WebAuthnUserAccount find(long id) {
+        WebAuthnUserAccount account = accountRepository.findOne(id);
+        if (account == null) {
+            return null;
+        }
+        return accountRepository.detach(account);
+    }
+
+    @Transactional(readOnly = true)
+    public List<WebAuthnUserAccount> findBySubjectAndRealm(String subject, String realm) throws NoSuchUserException {
+        List<WebAuthnUserAccount> accounts = accountRepository.findBySubjectAndRealm(subject, realm);
+        List<WebAuthnUserAccount> result = new LinkedList<>();
```

Comment:
we should avoid copying results into a new list. Use stream.map to detach entities

Reply:
Fixed in 1087830eaa48c4faa46627f1b7c848eab27c8805